### PR TITLE
Empty results in filtering

### LINKED ISSUE #236: When filtering returns no results show that to the user
image

Add `EmptyState` which shows `No files or directories found`.

As a bonus add a cross button which removes the current filter as in the hosts selector:

image

### PR DESCRIPTION
**Pull request by Abdelra6**
closes #236 

I added clear option in the `InputSearch` and clear Search button if the no result matched the filter

### CODE REVIEW

File: src/navigator-card-body.jsx [hunk 1]
```diff
@@ -280,7 +287,24 @@ export const NavigatorCardBody = ({
                 <Spinner />
             </Flex>
         );
-    if (isGrid) {
+    const clearFilter = () => {
+        setCurrentFilter("");
+    };
+    if (sortedFiles.length === 0) {
```
Comment (allisonkarlitskaya) [inline src/navigator-card-body.jsx:293]: I feel like this is probably also going to show on empty folders, even if there's no search term?

Probably we should restrict this to showing with non-empty search.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
hi!  Thanks for the patch!

Can you please squash the two commits into one and add a commit message describing what you're doing, with a `Closes #` tag?  Also: see one comment below.

@garrett normally does UI reviews but he doesn't work on Fridays.

File: src/navigator-card-body.jsx [hunk 2]
```diff
@@ -280,7 +287,24 @@ export const NavigatorCardBody = ({
...
+        return (
+            <EmptyState>
+                <EmptyStateHeader
+                  titleText="No results found" headingLevel="h4"
+                  icon={<EmptyStateIcon icon={SearchIcon} />}
+                />
+                <EmptyStateFooter>
+                    <EmptyStateActions>
+                        <Button variant="link" onClick={clearFilter}>Clear search</Button>
```
Comment (jelly) [inline src/navigator-card-body.jsx:302]: The clearing of the search filter should happen in the SearchInput

https://www.patternfly.org/components/text-input-group#utilities-and-icon

File: src/navigator-card-body.jsx [hunk 3]
```diff
@@ -280,7 +287,24 @@ export const NavigatorCardBody = ({
...
         );
-    if (isGrid) {
+    const clearFilter = () => {
+        setCurrentFilter("");
+    };
+    if (sortedFiles.length === 0) {
+        return (
+            <EmptyState>
+                <EmptyStateHeader
+                  titleText="No results found" headingLevel="h4"
```
Comment (jelly) [inline src/navigator-card-body.jsx:297]: Please make this translatable:

```suggestion
                  titleText={_("No results found")} headingLevel="h4"
```

File: src/navigator-card-body.jsx (hunk 1 above)
Comment (Abdelra6) [reply src/navigator-card-body.jsx]: You're right. I made a new commit. Could you please look into it?

**Review by garrett — CHANGES_REQUESTED** [review]
Thanks for this PR!

Unfortunately, it changes the size of the input. It should not.

Here's a before and after. You can see the [x] button pushes the input area to the left:

image

I think this is what @jelly meant in his review above https://github.com/cockpit-project/cockpit-navigator/pull/240#discussion_r1475773310

**Comment by Abdelra6** [comment]
The [x] button wasn't there before jelly's review. the review was the adding of this button.  Anyways thanks for the review

**Comment by garrett** [comment]
> The [x] button wasn't there before jelly's review. the review was the adding of this button. Anyways thanks for the review

Yep, I understand that. It should show up on the inside, instead of the outside though. Thanks!

**Pull request closed without merging**